jems-testing/src/test/java/org/dmfs/jems/hamcrest/matchers/CharSequenceMatcherTest.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.hamcrest.matchers;
+
+import org.hamcrest.Description;
+import org.hamcrest.StringDescription;
+import org.junit.Test;
+
+import static org.dmfs.jems.hamcrest.matchers.CharSequenceMatcher.validCharSequence;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+
+/**
+ * @author Marten Gajda
+ */
+public class CharSequenceMatcherTest
+{
+    @Test
+    public void testCharSequenceMatches() throws Exception
+    {
+        assertTrue(validCharSequence("").matches(""));
+        assertTrue(validCharSequence("1").matches("1"));
+        assertTrue(validCharSequence("1234").matches("1234"));
+    }
+
+
+    @Test
+    public void testCharSequenceChatAtMismatch() throws Exception
+    {
+        // test charAt does not throw
+        assertFalse(validCharSequence("123").matches(new TestSequence("123")
+        {
+            @Override
+            public char charAt(int i)
+            {
+                return i != 1 ? super.charAt(i) : 'x';
+            }
+        }));
+
+        Description description = new StringDescription();
+    }
+
+
+    @Test
+    public void testCharSequenceChatAtNoException1() throws Exception
+    {
+        // test charAt does not throw
```

Comment:
Comment is wrong here as well. Not sure about the test method names..

Reply:
I usually don't care too much about test method names. If a test too complex to understand it from just looking at it there should be a comment that describes what's being tested. Long and clumsy method names don't help much in that case.

This particular name should be read as:

`test CharSequence.CharAt - No Exception (version) 1`

(I just fixed the typo in the name in another commit, in case that's what you were referring to)